src/authorization/WebAclAuthorizer.ts

```diff
@@ -146,7 +166,12 @@ export class WebAclAuthorizer extends Authorizer {
     }
 
     // Check if any collected authorization block allows the specific agent
-    return auths.some((term): boolean => this.hasAccess(agent, term, acl));
+    for (const auth of auths) {
+      if (await this.hasAccess(agent, auth, acl)) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
I wouldn't do this serially.

Perhaps change `hasAccess` to `checkAccess` and make it throw on failure instead; then we can `Promise.any`.

Reply:
Hi @RubenVerborgh,
how can I use Promise.any? It looks like it's a very recent feature (ES2021), which is supported only in NodeJS versions >=15.0.0 ([changelog reference](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V15.md#v8-86---35415)). I found [this npm package](https://www.npmjs.com/package/promise.any) but I didn't manage to get the code work with it.

Reply:
Hmm, in that case a `Promise.all` should do.

Reply:
Or actually, given that this pattern also occurs below, we can abstract this into a function `hasSome`, which does the loop.

Reply:
Fixed. I'm planning to update the hasSome implementation so that it uses Promise.all